- Drop forced -O2
- Only Linux requires libnuma for MySQL
- Use C++ compiler for linking test

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -30,12 +30,6 @@ AC_MSG_RESULT($host_os)
 AC_MSG_CHECKING(hardware)
 AC_MSG_RESULT($host_cpu)
 
-dnl initial checks; tweakings to CFLAGS and LDFLAGS
-dnl see final checks for tweakings to LIBS
-if test "x$ac_cv_prog_gcc" = xyes ; then
-  CFLAGS="-O2 ${CFLAGS}"
-fi
-
 AC_CHECK_FUNCS([strlcpy])
 
 dnl: shared object handling
@@ -403,10 +397,6 @@ AC_ARG_ENABLE(mysql,
     CFLAGS="$CFLAGS $MYSQL_CFLAGS"
     AC_CHECK_HEADERS([mysql.h mysql/mysql.h])
     CFLAGS="$_save_CFLAGS"
-
-    AC_CHECK_LIB([numa], [numa_bind], [], [AC_MSG_ERROR([
-      ERROR: missing libnuma. Requirement for building MySQL.
-    ])])
     ;;
   no)
     AC_MSG_RESULT(no)
@@ -1176,10 +1166,9 @@ AC_LANG_PUSH([C++])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],[AC_MSG_RESULT(yes)],[AC_MSG_ERROR([No C++ compiler found])])
 AC_LANG_POP([C++])
 
-LIBS="-lcdada -lstdc++ ${LIBS}"
 AS_CASE(["$WITH_EXTERNAL_DEPS"],
   [yes], [],
-  [no], [AC_CHECK_LIB([cdada], [cdada_get_ver], [], [AC_MSG_ERROR([Could not find libcdada])])],
+  [no], [AC_LANG_PUSH([C++]) AC_CHECK_LIB([cdada], [cdada_get_ver], [], [AC_MSG_ERROR([Could not find libcdada])]) AC_LANG_POP([C++])],
   [])
 
 dnl set debug level
